### DISCUSSION TITLE
Publishing of Workflows

### DISCUSSION THREAD

**Discussion by basicSoloMan** [elsa-workflows/elsa-core#4337] [Q&A]
I noticed that a finished activity is added to the previous versions of a workflow when publishing a new one. I have workflows in suspended status with an activity that's supposed to wait for a couple of days then execute again, and it seems that these instances are not resuming as expected after the period it's supposed to wait. My question is what happens to workflow instances with a suspended status when a new version of that workflow is published? Will that version of the workflow still execute as expected?

**Comment by sfmskywalker** [comment]
Hello @basicSoloMan,

You're right in noting that a suspended workflow instance should resume without issues, regardless of newer versions of the workflow definition being published.

If suspended workflow instances aren't resuming correctly after publishing a new version, it might be a bug. I haven't encountered this issue, but we'd like to look into it.

Can you provide a step-by-step guide to replicate this behavior? It will help us identify and address the problem.

Also, which version of Elsa are you using?
